core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -154,18 +115,58 @@ public void close() throws IOException {
         IOUtils.close(closeables);
     }
 
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        IngestMetadata ingestMetadata = event.state().getMetaData().custom(IngestMetadata.TYPE);
+        if (ingestMetadata == null) {
+            return;
+        }
+
+        Map<String, Pipeline> pipelines = new HashMap<>();
+        for (Map.Entry<String, Map<String, Object>> entry : ingestMetadata.getPipelines().entrySet()) {
+            String id = entry.getKey();
+            try {
+                @SuppressWarnings("unchecked")
+                Map<String, Object> pipelineConfigCopy = (Map<String, Object>) deepCopy(entry.getValue());
+                pipelines.put(id, constructPipeline(id, pipelineConfigCopy));
```

Comment:
given where it ended up being called, I think that removing properties from config is not that useful anymore? maybe we should have two versions of the method, one to validate and one called here so we avoid the deep copy? in theory things should never fail at this point right?